server/route.go

```diff
@@ -1162,7 +1162,7 @@ func (c *client) removeRemoteSubs() {
 	c.mu.Lock()
 	srv := c.srv
 	subs := c.subs
-	c.subs = make(map[string]*subscription)
+	c.subs = nil
```

Comment:
Everywhere in this PR where I set `c.subs = nil`, I could, like we were doing here, create an empty map. But I believe that we check for connection not being closed when accessing (at least setting a value to) c.subs.